src/main/java/com/arangodb/springframework/repository/query/ArangoResultConverter.java

```diff
@@ -112,18 +112,13 @@ public ArangoResultConverter(final ArangoParameterAccessor accessor, final Arang
 	 * @return result in desired type
 	 */
 	public Object convertResult(final Class<?> type) {
-		try {
-			if (type.isArray()) {
-				return TYPE_MAP.get("array").invoke(this);
-			}
-			if (!TYPE_MAP.containsKey(type)) {
-				return getNext(result);
-			}
-			return TYPE_MAP.get(type).invoke(this);
-		} catch (final Exception e) {
-			e.printStackTrace();
-			return null;
+		if (type.isArray()) {
+			return TYPE_MAP.get("array").invoke(this);
+		}
+		if (!TYPE_MAP.containsKey(type)) {
+			return getNext(result);
 		}
+		return TYPE_MAP.get(type).invoke(this);
```

Comment:
Either throws declaration or something like:

```suggestion
		try {
			if (type.isArray()) {
				return TYPE_MAP.get("array").invoke(this);
			}
			if (!TYPE_MAP.containsKey(type)) {
				return getNext(result);
			}
			return TYPE_MAP.get(type).invoke(this);
		} catch (InvocationTargetException | IllegalAccessException e) {
			throw new ArangoUncategorizedException("Unable to convert the result.", e);
		}
```